### PR TITLE
Notes import/export moved to settings

### PR DESCRIPTION
**Pull request by Merkost**

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/notes/pro/activities/SettingsActivity.kt [hunk 1]
```diff
@@ -257,4 +288,62 @@ class SettingsActivity : SimpleActivity() {
...
+            toast(R.string.no_entries_for_exporting)
+        } else {
+            try {
+                val outputStream = contentResolver.openOutputStream(uri)!!
+
+                val jsonString = Json.encodeToString(notes)
+                outputStream.use {
+                    it.write(jsonString.toByteArray())
+                }
+                Log.e("EXPORTED", jsonString)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/notes/pro/activities/SettingsActivity.kt:316]: guess you left it by accident

File: app/src/main/res/layout/activity_settings.xml [hunk 2]
```diff
@@ -346,6 +346,46 @@
                     android:text="@string/display_success_message" />
 
             </RelativeLayout>
+
+            <include
+                android:id="@+id/settings_general_settings_divider"
+                layout="@layout/divider" />
+
+            <TextView
+                android:id="@+id/settings_migrating_label"
```
Comment (tibbi) [inline app/src/main/res/layout/activity_settings.xml:355]: add the label at onResumes text setter arraylist to make sure it uses the proper color

File: app/src/main/kotlin/com/simplemobiletools/notes/pro/activities/SettingsActivity.kt [hunk 3]
```diff
@@ -1,26 +1,55 @@
...
+    private val notesFileType = "application/json"
+
+    private val getContent = registerForActivityResult(ActivityResultContracts.GetContent()) { uri ->
+        if (uri != null) {
+            toast(R.string.importing)
+            importNotes(uri)
+        }
+    }
+
+    private val saveDocument = registerForActivityResult(ActivityResultContracts.CreateDocument(notesFileType)) { uri ->
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/notes/pro/activities/SettingsActivity.kt:40]: do these really have to be values, not some functions? And move them lower, all activities start with the lifecycle functions like onCreate etc

**Comment by tibbi** [comment]
there should be a renaming dialog right after clicking on Export notes, just like in exporting the galleries app settings, favorites etc

**Comment by tibbi** [comment]
if I export a textual note and a checklist with a couple items, then delete the checklist note and try reimporting the file, I get an error that importing failed and nothing gets imported. If I open the file, the checklist note is the first one there.

File: app/src/main/kotlin/com/simplemobiletools/notes/pro/activities/SettingsActivity.kt (hunk 3 above)
Comment (Merkost) [reply app/src/main/kotlin/com/simplemobiletools/notes/pro/activities/SettingsActivity.kt:40]: Yeah, that's exactly how new ActivityResult is working

**Comment by Merkost** [comment]
> if I export a textual note and a checklist with a couple items, then delete the checklist note and try reimporting the file, I get an error that importing some entries failed and nothing gets imported. If I open the file, the checklist note is the first one there.

It was importing, but only the pager was not updated. App restart needed after import. I'll fix it now. And some entries failed because some documents with the same title already exist.

**Comment by tibbi** [comment]
seems to work good now, but if no note has been added, instead of `Importing failed` lets show `No new items have been found` to make it a bit clearer

**Pull request merged**